src/main/webapp/script.js

```diff
@@ -23,24 +23,27 @@ google.setOnLoadCallback(getChart)
 
 async function getYouTubeComments() { 
   const urlInput = document.getElementById('link-input');
-  const url = cleanseUrl(urlInput.value);
+  const url = extractYouTubeUrl(urlInput.value);
   const response = await fetch("/YouTubeComments?url="+url);
   const comments = await response.json();
   return comments;
 }
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
+ * Extracts video ID from the full URL
```

Comment:
Since your PR mentions that you emulated your parsing from a method you found on StackOverflow, you should include that link within the javadoc for this method.